modules/is_analytics_worker_master/build.sh

```diff
@@ -0,0 +1,165 @@
+then
+  mkdir ${distribution_path}/updates
+fi
+
+# Getting update status
+# 0 - first/last update successful
+# 1 - Error occurred in last update
+# 2 - In-place has been updated
+# 3 - conflicts encountered in last update
+status=0
+if [[ -f ${distribution_path}/updates/status ]]
+then
+  status=$(cat ${distribution_path}/updates/status)
+fi
+
+# Move into binaries of installation directory
+cd ${install_path}/bin
+
+# Run in-place update
+if [[ ${status} -eq 0 ]] || [[ ${status} -eq 1 ]] || [[ ${status} -eq 2 ]]
+then
+  ./update_linux --verbose 2>&1 | tee ${install_path}/bin/output.txt
+  update_status=${PIPESTATUS[0]}
+elif [[ ${status} -eq 3 ]]
+then
+  ./update_linux --verbose --continue 2>&1 | tee ${install_path}/bin/output.txt
+  update_status=${PIPESTATUS[0]}
+
+  # Handle user running update script without resolving conflicts
+  if [[ ${update_status} -eq 1 ]]
+  then
+    echo "Error occurred while attempting to resolve conflicts."
+    exit 1
+  fi
+else
+  echo "status file is invalid. Please delete or clear file content."
+  exit 1
+fi
+
+# Update status
+echo ${update_status} > ${distribution_path}/updates/status
+if [[ ${update_status} -eq 0 ]]
+then
+  echo
+  echo "Update completed successfully."
+  copy_to_agent "updated"
+elif [[ ${update_status} -eq 3 ]]
+then
+  echo "Conflicts encountered. Please resolve conflicts and run the update script again."
+else
+  echo "Update error occurred. Stopped with exit code ${update_status}"
+  exit ${update_status}
+fi
+
+# Get list of merged files
+if [[ ${update_status} -eq 0 ]] # If update is successful
+then
+  sed -n '/files./,/Successfully/p' ${install_path}/bin/output.txt > ${install_path}/bin/merged_files.txt
+elif [[ ${update_status} -eq 3 ]] # If conflicts were encountered during update
+then
+  sed -n '/files./,/Merging/p' ${install_path}/bin/output.txt > ${install_path}/bin/merged_files.txt
+fi
+
+if [[ -s ${install_path}/bin/merged_files.txt ]]
+then
+  sed -i '1d' ${install_path}/bin/merged_files.txt # Remove first line from file
+  sed -i '$ d' ${install_path}/bin/merged_files.txt # Remove last line from file
+
+  while read -r line; do
+    filepath=${line##*${product}-${product_version}/}
+    template_file=${puppet_env}/modules/is_master/templates/carbon-home/${filepath}.erb
+    if [[ -f ${template_file} ]]
+    then
+      updated_templates+=(${template_file})
+    fi
+  done < ${install_path}/bin/merged_files.txt
+
+  # Display template files to be changed
+  if [[ -n ${updated_templates} ]]
+  then
+    DATE=`date +%Y-%m-%d`
+    update_file_name="update_${DATE}.log"
+    echo
+    echo "Update has made changes to the following files. Please update the templates accordingly before running the next update." | tee -a ${distribution_path}/updates/${update_file_name}
+    printf '%s\n' "${updated_templates[@]}" | tee -a ${distribution_path}/updates/${update_file_name}
+  fi
+fi
+
+# Clean files
+rm ${install_path}/bin/output.txt ${install_path}/bin/merged_files.txt
```

Comment:
Add an empty line to the EOF

Reply:
Resolved https://github.com/wso2/puppet-is/pull/62/commits/d115f04492787980146f632a774a7a6d1f4f5466